ui/raidboss/popup-text.js

```diff
@@ -426,6 +426,11 @@ class PopupText {
     let suppress = 'suppressSeconds' in trigger ? ValueOrFunction(trigger.suppressSeconds) : 0;
     if (trigger.id && suppress > 0)
       this.triggerSuppress[trigger.id] = now + suppress * 1000;
+    let promise = 'promise' in trigger ? promise : () => {
```

Comment:
I think you're missing `trigger.`, i.e. `let promise = 'promise' in trigger ? trigger.promise : () => {`

Reply:
Yes, you're correct. Fixing that now.

Reply:
Also it seems like `trigger.promise` must always be a function that returns a promise.  (This seems reasonable! I don't think it makes sense here to have a raw promise as the value associated with the `promise` key.)  Can you verify that `trigger.promise` is a function and later on that what it returns is an instanceof Promise, and throw a trigger error if not?

Reply:
I assume that the errors in the latest commit are what you're referring to when you say 'trigger error'?